### PR TITLE
[DOCS] add tutorial for large dataset DataLoader

### PR DESCRIPTION
**Pull request by jasminerienecker**
A follow up tutorial for this review https://github.com/Nixtla/neuralforecast/pull/1049

### CODE REVIEW

File: nbs/examples/Large_Datasets_Tutorial.ipynb [hunk 1]
```diff
@@ -0,0 +1,726 @@
+{
```
Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <p>Please add a cell below this one with the following:</p>

<pre class="ql-syntax" spellcheck="false">logging.getLogger('pytorch_lightning').setLevel(logging.ERROR)
os.environ['NIXTLA_ID_AS_COL'] = '1'
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    import os
</pre><p>Please also use a temporary directory here to avoid leaving files behind</p>

<pre class="ql-syntax" spellcheck="false">import logging
import os
import tempfile
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    train.to_parquet('./data', partition_cols=['unique_id'], index=False)
</pre><p>Please define a temporary directory here and use its name</p>

<pre class="ql-syntax" spellcheck="false">tmpdir = tempfile.TemporaryDirectory()
train.to_parquet(tmpdir.name, partition_cols=['unique_id'], index=False)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #9.    files_list = [f"data/{dir}" for dir in os.listdir('./data')]
</pre><p>Please also use the tempdir here</p>

<pre class="ql-syntax" spellcheck="false">files_list = [f"data/{dir}" for dir in os.listdir(tmpdir.name)]
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #12.    nf = NeuralForecast(models=models, freq='M')
</pre><p>Please use <code>freq='ME'</code> here to address possible deprecation warnings from pandas</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    predictions = nf.predict(df=pred_df, futr_df=futr_df, static_df=static).reset_index()
</pre><p>With the id as col env variable the reset_index won't be necessary anymore</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <p>If you update utilsforecast to 0.2.0 this can now be done like this:</p>

<pre class="ql-syntax" spellcheck="false">&nbsp;&nbsp;&nbsp; evaluate(
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; predictions.merge(target.drop(["trend", "y_[lag12]"], axis=1), on=['id_col', 'ds']),
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; metrics=[mae, rmse, smape],
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; id_col='id_col',
        agg_fn='mean',
&nbsp;&nbsp;&nbsp; )
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

**Comment by jmoralez** [comment]
Looks great! Just some minor suggestions

File: nbs/examples/Large_Datasets_Tutorial.ipynb (hunk 1 above)
Comment (jasminerienecker) [reply nbs/examples/Large_Datasets_Tutorial.ipynb:1]: sounds good - just flagging I've set the file list to be "(tmpdir.name)/{dir}" instead

File: nbs/examples/Large_Datasets_Tutorial.ipynb [hunk 2]
```diff
@@ -0,0 +1,691 @@
...
```
Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <p>Can you please change the last sentence from</p><blockquote>only one timeseries is ever loaded</blockquote><p>to</p><blockquote>only one batch is ever loaded</blockquote>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (jmoralez) [inline nbs/examples/Large_Datasets_Tutorial.ipynb:1]: <p>Can you please add a markdown cell below this one with the following spark example?</p><p> </p><blockquote>You can also create this directory structure from a spark dataframe with the following:</blockquote>

<pre class="ql-syntax" spellcheck="false">spark.conf.set("spark.sql.parquet.outputTimestampType", "TIMESTAMP_MICROS")
(
&nbsp; spark_df
&nbsp; .repartition(id_col)
&nbsp; .sortWithinPartitions(id_col, time_col)
&nbsp; .write
&nbsp; .partitionBy(id_col)
&nbsp; .parquet(out_dir)
)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/1074/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Review by jmoralez — APPROVED** [review]
Thanks!
